enhancements/multi-arch/heterogenous-architecture.md

```diff
@@ -0,0 +1,270 @@
+- Another option would be to mandate an x86 worker node be present for Hypershift and Hypershift would select the x86 node for hosted cluster deployment
+- When creating the nodepool, the defaults for instances types and AMIs are x86 specific. The instance types can be overridden but not the AMIs. An option needs to be added to the create cluster command to override AMIs
+
+### Console
+
+As of today console filters content based on control plane architecture. This needs to be entirely removed or made configurable for heterogenous clusters
+
+### Install
+
+For Phase 1, the way to add more workers to the cluster is to handcraft a MachineSet manifest file for a worker node and specify the architecture specific details like instance type, image references and deploy the machineset to get a worker node. This can be pseudo automated
+further by creating the manifests using `openshift-install create manifests` and modifying or adding worker manifest files. Eventually the installer should support being able to declare differing compute and control plane sections in the install-config mainly allowing
+architectures to be different which is not allowed today.
+
+### Open Questions
+
+Definition of GA - Plan is to GA phase 1 while documenting the limitations. This is easier than creating support exceptions or limiting payload access.
+
+### Test Plan
+
+Testing for Phase 1:
+- QE would need to test heterogenous nightly payloads before the entry is added to Cincinnati. This implies that the heterogenous payloads will be released a bit later than the homogenous payloads
+- Testing scenarios would include machine set scaling, upgrades involving ARM nodes and OVN testing on heterogenous clusters
+- New release payload acceptance jobs on heterogenous clusters
+
+Testing for Phase 2:
+- With the addition of Power and Z architectures, there will be more permutations to test. We would need to figure out the absolute essential set of permutations and the tests to run on those
+- Imagestream conformance tests
+
+### Graduation Criteria
+
+Phase 1 will have a pre-4.11 dev preview and 4.11 GA with documented limitations
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+This enhancement does not describe removing a feature of Openshift, this section is not applicable.
+
+### Upgrade / Downgrade Strategy
+
+The upgrade strategy is described in the design details section above. The important changes here being a separate channel for the heterogenous releases and also the work needed to support homogenous release -> heterogenous release and vice versa in the future.
+
+### Version Skew Strategy
+
+TBD
+
+### Operational Aspects of API Extensions
+
+There will be minor changes to the cluster or infrastructure APIs for supporting upgrades and possibly Hypershift
```

Comment:
imagestream apis may change slightly as well, as part of introducing support for manifestlist import behavior

Reply:
added.